tests/xml_time_format_tests.cpp

```diff
@@ -49,3 +51,56 @@ TEST_CASE("xml_time_format") {
     CHECK_THAT(xml.str(), EqualsXmlFile("write_time_format"));
   }
 }
+
+TEST_CASE("xml_time_format_sadm") {
+  // has lstart/lduration but parsing without FrameHeader so assumes total timeReference
+  CHECK_THROWS(parseXml("xml_parser/time_format_sadm.xml"));
+  auto document = parseXml("xml_parser/time_format_sadm.xml",
+                           xml::ParserOptions::permit_time_reference_mismatch);
+
+  SECTION("parser") {
+    SECTION("DirectSpeakers") {
+      auto programme = document->lookup(parseAudioProgrammeId("APR_1001"));
+      REQUIRE(programme);
+      REQUIRE(programme->get<Start>().get() == FractionalTime{1, 48000});
+      REQUIRE(programme->get<End>().get() == FractionalTime{123, 48000});
+    }
+
+    SECTION("DirectSpeakers") {
+      auto channelFormat =
+          document->lookup(parseAudioChannelFormatId("AC_00011001"));
+      REQUIRE(channelFormat);
+      auto firstBlockFormat = *(
+          channelFormat->getElements<AudioBlockFormatDirectSpeakers>().begin());
+      REQUIRE(firstBlockFormat.has<Rtime>());
+      REQUIRE(firstBlockFormat.get<Rtime>().get() == FractionalTime{1, 2});
+      REQUIRE(firstBlockFormat.has<Duration>());
+      REQUIRE(firstBlockFormat.get<Duration>().get() == FractionalTime{5, 4});
+    }
+
+    SECTION("Objects") {
+      auto channelFormat =
+          document->lookup(parseAudioChannelFormatId("AC_00031001"));
+      REQUIRE(channelFormat);
+      auto firstBlockFormat =
+          *(channelFormat->getElements<AudioBlockFormatObjects>().begin());
+      REQUIRE(firstBlockFormat.has<Rtime>());
+      REQUIRE(firstBlockFormat.get<Rtime>().get() == FractionalTime{1, 2});
+      REQUIRE(firstBlockFormat.has<Duration>());
+      REQUIRE(firstBlockFormat.get<Duration>().get() == FractionalTime{5, 4});
+    }
+  }
+
+  // Not sure why this test fails when the write_time_format_sadm.accepted.xml matches the xml string perfectly
```

Comment:
it's missing the frame header, so writes rtimes rather than ltimes. I guess we just can't write documents correctly that are only parseable with with `permit_time_reference_mismatch`?